### DISCUSSION TITLE
Multiple slot templates and layout changes

### DISCUSSION THREAD

**Discussion by Miksser** [nuxt/nuxt#16804] [Questions]
Hello! 

Is it possible to use multiple named slot templates in one component and still switch the layout? 
I need this to make several themes for the site, but do not change the components themselves. 

Fundamentally, I didn't find a way to use setPageLayout() and <NuxtLayout name="custom"> right now

**Comment by sevillaarvin** [comment]
Hi, can you clarify what you mean? I believe layout is set on a per page basis, not per component, or perhaps I misunderstood you?
